src/diffusers/models/embeddings.py

```diff
@@ -78,6 +78,53 @@ def get_timestep_embedding(
     return emb
 
 
+def aryan_get_3d_sincos_pos_embed(
```

Comment:
We could compare this and the numpy implementation on the same inputs and see if there's any divergence. Usually a good start to try to localize if there are any problems.

Reply:
It seems to match for all cases except the 3D sincos ones, based on my test code above. I'll give it a look again soon